test/fw/ptl/utils/plugins/ptl_test_info.py

```diff
@@ -213,8 +234,12 @@ def _gen_ts_tree(self, suite):
                     # not a test case, ignore
                     continue
                 tcd['doc'] = str(tc.__doc__)
-                tctags = sorted(set(tstags + getattr(tc, TAGKEY, [])))
+                tc_req = getattr(tc, REQUIREMENTS_KEY, {})
+                eff_tc_req = get_eff_requirements(ts_req, tc_req)
+                if eff_tc_req:
+                    tcd['requirements'] = eff_tc_req
```

Comment:
I don't think we need to put this under if condition, we should simply `requirements` key irrespective of whether the test has a requirement or not and by doing that we will get uniform JSON output.
If we add requirements key in JSON based on condition then some JSON output will have requirement key and some not which makes non-uniform output. What do you say?